comp/otelcol/otlp/configcheck/configcheck.go

```diff
@@ -75,7 +75,15 @@ var intConfigs = map[string]struct{}{
 
 // IsEnabled checks if OTLP pipeline is enabled in a given config.
 func IsEnabled(cfg config.Reader) bool {
-	return hasSection(cfg, coreconfig.OTLPReceiverSubSectionKey)
+	if cfg.Get("otlp_config.enabled") == true {
+		return true
+	}
```

Comment:
If explicitly set to false by the user we should not enable OTLP I think

```suggestion
	if cfg.Get("otlp_config.enabled") == true {
		return true
	} else if  cfg.IsConfigured("otlp_config.enabled") {
		return false
	}
```